src/lxml/classlookup.pxi

```diff
@@ -550,6 +550,16 @@ def set_element_class_lookup(ElementClassLookup lookup = None):
     u"""set_element_class_lookup(lookup = None)
 
     Set the global default element class lookup method.
+
+    .. warning::
+
+        Setting the default element class to something other than a
+        :class:`ParserBasedElementClassLookup` will prevent
+        :meth:`~lxml.etree.XMLParser.set_element_class_lookup` from working.
+
+        Amongst other issues, this will break :mod:`lxml.html` and
+        :mod:`lxml.objectify`'s extensions as they rely on lookups configured
+        on their respective parsers.
```

Comment:
```suggestion
    .. warning::

        This should only be changed by applications, not by library packages.
        In most cases, parser specific lookups should be preferred,
        which can be configured via
        :meth:`~lxml.etree.XMLParser.set_element_class_lookup`
        (and the same for HTML parsers).

        Globally replacing the element class lookup by something other than a
        :class:`ParserBasedElementClassLookup` will prevent parser specific lookup
        schemes from working. Several tools rely on parser specific lookups,
        including :mod:`lxml.html` and :mod:`lxml.objectify`.
```

Reply:
My only nitpick with this is even for applications it can be be a risky bet or a dead end and should be considered very carefully.

Though of course the application author / user will likely quickly realise they've broken`lxml.html` or `lxml.objectify` if they were using those (obviously), the inter-relation can be quite non-obvious.

A specific admonitions that libraries should never ever touch this is definitely a good idea tho.